model_analyzer/result/sorted_results.py

```diff
@@ -91,37 +98,66 @@ def top_n_results(self, n):
 
         Returns
         -------
-        list of ModelResults
+        list of RunConfigResults
             The n best results for this model,
             must all be passing results
         """
-
         if len(self._passing_results) == 0:
             logger.warning(
                 f"Requested top {n} configs, but none satisfied constraints. "
                 "Showing available constraint failing configs for this model.")
 
             if n == -1:
-                return heapq.nsmallest(len(self._failing_results),
-                                       self._failing_results)
+                return self._failing_results
             if n > len(self._failing_results):
                 logger.warning(
                     f"Requested top {n} failing configs, "
                     f"but found only {len(self._failing_results)}. "
                     "Showing all available constraint failing configs for this model."
                 )
-            return heapq.nsmallest(min(n, len(self._failing_results)),
-                                   self._failing_results)
+
+            result_len = min(n, len(self._failing_results))
+            return self._failing_results[0:result_len]
 
         if n == -1:
-            return heapq.nsmallest(len(self._passing_results),
-                                   self._passing_results)
+            return self._passing_results
         if n > len(self._passing_results):
             logger.warning(
                 f"Requested top {n} configs, "
                 f"but found only {len(self._passing_results)} passing configs. "
                 "Showing all available constraint satisfying configs for this model."
             )
 
-        return heapq.nsmallest(min(n, len(self._passing_results)),
-                               self._passing_results)
+        result_len = min(n, len(self._passing_results))
+        return self._passing_results[0:result_len]
+
+    def _find_existing_run_config(
```

Comment:
this name implies it is finding a run config. Is it?

Reply:
Yes. Well sort of...it's getting the RunConfigResult, which is the RunConfig + other necessary stuff  (comparator + constraints). 

I omitted the extra result part to try and not have so many things with result in the name. Type checking indicates what this actually is, so I thought the brevity was okay in this case.

Reply:
Like we discussed I've removed the passing/failing lists and now create them only when needed. I've also updated all the names to reflect this and added unit tests to cover the case where a result goes from failing to passing.